asf_public_discourse_home_decarbonisation/analysis/mse/eda_mse_category_data.py

```diff
@@ -0,0 +1,311 @@
+where
+<category>: category or sub-forum to be analysed, e.g. "energy", "lpg-heating-oil-solid-other-fuels", etc.
+<collection_date_time>: data collection date/time in the format YYYY_MM_DD
+"""
+
+# ## Package imports
+import warnings
+
+# Ignoring warnings from Pandas
+warnings.simplefilter(action="ignore")
+
+import pandas as pd
+from datetime import datetime
+import matplotlib.pyplot as plt
+import altair as alt
+import os
+import asf_public_discourse_home_decarbonisation.config.plotting_configs as pc
+from asf_public_discourse_home_decarbonisation.getters.mse_getters import get_mse_data
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+import argparse
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def add_datetime_variables(mse_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Transform the datetime variable into a datetime object and add year and year_month variables.
+    Args:
+        mse_data (pd.DataFrame): dataframe with MSE data
+    Returns:
+        pd.DataFrame: Pandas DataFrame with datetime variables added
+    """
+    mse_data["datetime"] = mse_data["datetime"].apply(
+        lambda x: datetime.strptime(x, "%Y-%m-%dT%H:%M:%S+00:00")
+    )
+    mse_data["year"] = mse_data["datetime"].apply(lambda x: x.year)
+    mse_data["year_month"] = mse_data["datetime"].dt.to_period("M")
+
+    return mse_data
+
+
+def create_statistics_from_data(mse_data: pd.DataFrame):
+    """
+    Create and prints statistics from the MSE data.
+
+    Args:
+        mse_data (pd.DataFrame): _description_
+    """
+    if len(mse_data) == len(mse_data.drop_duplicates()):
+        logger.info("There are no duplicated posts in the data.")
+    else:
+        logger.warning(
+            f"There are duplicated posts in the data.\nNumber instances: {len(mse_data)}\nNumber unique instances: {len(mse_data.drop_duplicates())}"
+        )
+
+    logger.info(f"Categories in data: {mse_data['category'].unique()}")
+
+    logger.info(
+        f"Number of instances (posts & replies) collected for category '{category}': {len(mse_data)}"
+    )
+
+    number_original_posts = len(mse_data[mse_data["is_original_post"] == 1])
+    logger.info(
+        f"Number of original posts collected for category '{category}': {number_original_posts}"
+    )
+
+    logger.info(
+        f"Data collected ranges from {mse_data['datetime'].min()} to {mse_data['datetime'].max()}"
+    )
+
+
+def plot_number_posts_per_year(mse_data: pd.DataFrame, category: str):
+    """
+    Plots and saves a bar plot with number of posts per year.
+
+    Args:
+        mse_data (pd.DataFrame): Money Saving Expert data
+        category (str): MSE category/sub-forum
+    """
+    posts_per_year = mse_data.groupby("year")[["id"]].nunique()
+    posts_per_year.columns = ["number_posts"]
+    posts_per_year.reset_index(inplace=True)
+
+    chart = (
+        alt.Chart(posts_per_year)
```

Comment:
Basically what I was wondering is what you like about using altair over other packages such as seaborn or matplotlib?

Reply:
and for other plotting functions you seem to use matplotlib instead of altair.

Reply:
A lot of people in DSP are using Altair and I've been trying to learn it and use it too. I guess if you then create a Streamlit app, then you can have interactive Altair plots which looks way more cool than a static matplotlib plot :) sometimes Altair is a bit of a pain (e.g. you have too many bins for example, so you need to aggregate data further prior to plotting), so when that's the case I stick with Matplotlib.

I'd recommend these two Medium blogs by @VolcanoBlue13, @j-gillam and @LizzieIngram195:
- [Vega-Altair — a surprisingly powerful Python library for plotting interactive maps](https://medium.com/data-analytics-at-nesta/vega-altair-interactive-maps-d9c3bd77af50)
- [Creating Interactive Dashboards in Python: A Tutorial using Streamlit]
(https://medium.com/data-analytics-at-nesta/creating-interactive-dashboards-in-python-a-tutorial-using-streamlit-84ec94033a32)